### PR TITLE
Old medical tools 1: The Stethoscope, attempt 2, not based off of master this time

### PR DESCRIPTION
**Pull request by F-OS**
[ENHANCEMENT] [INPUT]
## About the PR 

This PR adds a stethoscope. It basically works as a slightly slower upgraded medical scanner for the lungs. It detects specific reagents, lung damage and respiratory failure, oxygen loss, and sleeping. This is very WIP and still needs sounds along with loop optimization and a few once-overs to make sure everything looks balanced and works well.

## Why's this needed? 
New toys for medical to play with are always fun, plus it could be useful for boosting RP on one or as something for Pizani to scam people with.

## Changelog

```
(u)Fosstar 
(+)Adds a stethoscope to the MD's medical bag that can be used to diagnose many conditions. Be careful though, your ears are at risk if you use it without the proper training or if you're a clown!
```

The sounds and the sprite have a massive amount of jank, any enhancements are extremely appreciated, also, this is the first serious addition I've ever made, so if I miss anything in terms of best practices in coding please let me know

### CODE REVIEW

File: code/obj/item/storage/med_chem.dm [hunk 1]
```diff
@@ -107,14 +107,30 @@
 	/obj/item/reagent_containers/emergency_injector/lexorin,\
 	/obj/item/reagent_containers/emergency_injector/synaptizine,\
 	/obj/item/device/analyzer/healthanalyzer)
-
+/*
+/obj/item/storage/firstaid/diagnostickit
```
Comment (ZeWaka) [inline code/obj/item/storage/med_chem.dm:111]: why is this commented out

File: code/obj/medicaldiagnosis.dm [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	name = "UM EXCUSE ME WHY ARE YOU SEEING THIS PLEASE MAKE AN ISSUE AAAAA"
+	desc = "*scream"
+	icon_state = "stethoscope" // todo
+	inhand_image_icon = "stethoscope" // nuke this awful sprite later
+	icon = 'icons/obj/medicaldiagnosis.dmi' // NOO YOU CAN'T JUST FILL YOUR CODE WITH TODO MARKERS YOU NEED TO ACTUALLY DO SPRITING AND NOT JUST MAKE CODERSPRITES NOO
+// haha procrastination machine go brrr
+/obj/item/medical/medicaldiagnosis/stethoscope
+	name = "stethoscope"
+	desc = "a disc-shaped resonator attached to two earpieces for figuring out if someone has consumption or is simply suffering from the vapors."
+	attack(mob/living/carbon/human/M as mob, mob/user as mob)
```
Comment (ZeWaka) [inline code/obj/medicaldiagnosis.dm:11]: why is this typed to humans? attack() is on all mobs.

File: code/obj/medicaldiagnosis.dm [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+								playsound(user, "sound/effects/sleepstethoscope.ogg", 40, 0, -6) // reusing because the sounds are similar
+							else if(M.reagents.has_reagent("epinephrine") || M.reagents.has_reagent("methamphetamine") || M.reagents.has_reagent("crank") || M.reagents.has_reagent("stimulants") || M.reagents.has_reagent("energydrink") )
+								boutput(user, "<span style=\"color:blue\">You hear fast breathing inside [(user != M) ? "[M]'s" : "your"] lungs.</span>")
+								playsound(user, "sound/effects/hyperventstethoscope.ogg", 40, 0, -6) // reusing because the sounds are similar
+							else if((20 < M.get_oxygen_deprivation()) && (M.get_oxygen_deprivation() < 80)) // pick from 3 stressed breath sounds when I feel like adding them
+								boutput(user, "<span style=\"color:blue\">You hear what sounds like hyperventilation [M.get_oxygen_deprivation() > 60 ? "with an irregular pattern" : ""] inside [(user != M) ? "[M]'s" : "your"] lungs.</span>") // oxygen low but they can still recover
+								if(M.get_oxygen_deprivation() < 60)
+									playsound(user, "sound/effects/hyperventstethoscope.ogg", 40, 0, -6)
```
Comment (ZeWaka) [inline code/obj/medicaldiagnosis.dm:77]: these sounds should be added to the sound cache

Comment (F-OS) [reply code/obj/medicaldiagnosis.dm:57]: Added, this should be resolved.

File: code/obj/medicaldiagnosis.dm (hunk 2 above)
Comment (F-OS) [reply code/obj/medicaldiagnosis.dm]: added an ishuman check and pulled out the scope, this should be fixed.

File: code/obj/item/storage/med_chem.dm (hunk 1 above)
Comment (F-OS) [reply code/obj/item/storage/med_chem.dm]: WIP stuff, removed.

**Comment by pali6** [comment]
Is this WIP or not? @Kyle2143 says it is but it seems fine to me?

**Comment by F-OS** [comment]
@pali6 technically yes but it’s in a playable state and assuming the code looks fine I don’t feel that it needs any other changes and can be merged as is. I defer to maintainer judgement on that however.

**Pull request merged**